### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-button

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2049]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-button component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-button can be found here: [sd-button Notes](https://www.notion.so/matuzo/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21).

------------------------------------------

## Comment
**Problem:** The [icon-only](https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-button-screenshots-sd-button--icon-only) buttons have no accessible name.
**Solution:** Provide an accessible name via slotted content, coming from the icon (needs a role=img for that), or aria-label.
<img width="1400" alt="Image" src="https://github.com/user-attachments/assets/6db980c1-9c33-4364-8722-5b2220bb0365" />

## Accessibility information
- Consider mentioning that button text should be unique.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by coraliefeil** [comment]
IMO the visually disabled button should show the tooltip not only on focus but also on hover so that also users with low vision can understand why it cannot be clicked.

See reference:
https://vimeo.com/582460935

Pls add this task to this ticket here. Thanks
